cmd/shfmt/testdata/scripts/flags.txt

```diff
@@ -92,6 +92,15 @@ stdin input-mksh
 shfmt -ln=auto -filename=input.mksh
 stdout 'coprocess'
 
+# Explicitly state language on parse errors
+stdin input-bash-arrays
+! shfmt -ln=auto -filename=input.sh
+stderr 'parsed as posix via -ln=auto'
+
+stdin input-bash-extglobs
+! shfmt -ln=auto -filename=input.sh
+stderr 'parsed as posix via -ln=auto'
```

Comment:
also add a test case without `-ln=auto`, which should behave the same as it's the default

Reply:
done!

Reply:
I repurposed the other one since its testing the same thing.

Reply:
SGTM